### ISSUE TITLE
Issue in recreating the results of an assign_model during predict_model in Anomaly Detection (lof)

### ISSUE THREAD

**Issue by praveenmec67** [pycaret/pycaret#2207]
I am currently working for an anomaly detection problem and I am using pycaret local outlier factor.

Generated Anomaly score for a data point using assign_model. When I try to predict the anomaly score (using predict_model) of the same data point the anomaly score is totally different.

Initially i used pycaret 2.3.4 with/without session_id in the setup().
Later i used pycaret 2.2.0 with session_id in the setup() (as suggested in the issue #505) but still facing the same issue.

Please find below the details of the OS.

2022-02-25 14:47:11,100:INFO:PyCaret Supervised Module
2022-02-25 14:47:11,101:INFO:ML Usecase: anomaly
2022-02-25 14:47:11,101:INFO:version 2.2.0
2022-02-25 14:47:11,101:INFO:Initializing setup()
2022-02-25 14:47:11,101:INFO:setup(target=None, ml_usecase=anomaly, available_plots={'tsne': 'Anomaly TSnE (3d)', 'umap': 'UMAP Dimensionality'}, train_size=0.7, test_data=None, preprocess=False, imputation_type=simple, iterative_imputation_iters=5, categorical_features=None, categorical_imputation=mode, categorical_iterative_imputer=lightgbm, ordinal_features=None, high_cardinality_features=None, high_cardinality_method=frequency, numeric_features=None, numeric_imputation=mean, numeric_iterative_imputer=lightgbm, date_features=None, ignore_features=None, normalize=False, normalize_method=zscore, transformation=False, transformation_method=yeo-johnson, handle_unknown_categorical=True, unknown_categorical_method=least_frequent, pca=False, pca_method=linear, pca_components=None, ignore_low_variance=False, combine_rare_levels=False, rare_level_threshold=0.1, bin_numeric_features=None, remove_outliers=False, outliers_threshold=0.05, remove_multicollinearity=False, multicollinearity_threshold=0.9, remove_perfect_collinearity=False, create_clusters=False, cluster_iter=20, polynomial_features=False, polynomial_degree=2, trigonometry_features=False, polynomial_threshold=0.1, group_features=None, group_names=None, feature_selection=False, feature_selection_threshold=0.8, feature_selection_method=classic, feature_interaction=False, feature_ratio=False, interaction_threshold=0.01, fix_imbalance=False, fix_imbalance_method=None, transform_target=False, transform_target_method=box-cox, data_split_shuffle=False, data_split_stratify=False, fold_strategy=kfold, fold=10, fold_shuffle=False, fold_groups=None, n_jobs=-1, use_gpu=False, custom_pipeline=None, html=True, session_id=123, log_experiment=False, experiment_name=None, log_plots=False, log_profile=False, log_data=False, silent=True, verbose=True, profile=False, display=None)
2022-02-25 14:47:11,101:INFO:Checking environment
2022-02-25 14:47:11,101:INFO:python_version: 3.8.5
2022-02-25 14:47:11,101:INFO:python_build: ('default', 'May 27 2021 13:30:53')
2022-02-25 14:47:11,101:INFO:machine: x86_64
2022-02-25 14:47:11,114:INFO:platform: Linux-3.10.0-1127.19.1.el7.x86_64-x86_64-with-glibc2.29
2022-02-25 14:47:11,115:INFO:Memory: svmem(total=811074142208, available=791762071552, percent=2.4, used=15170289664, free=486395641856, active=5767741440, inactive=5930586112, buffers=205799424, cached=309302411264, shared=2015916032, slab=303650631680)
2022-02-25 14:47:11,117:INFO:Physical Core: 16
2022-02-25 14:47:11,117:INFO:Logical Core: 32
2022-02-25 14:47:11,117:INFO:Checking libraries
2022-02-25 14:47:11,118:INFO:pd==1.3.3
2022-02-25 14:47:11,118:INFO:numpy==1.19.5
2022-02-25 14:47:11,118:INFO:sklearn==0.23.2
2022-02-25 14:47:11,368:INFO:xgboost==1.5.2
2022-02-25 14:47:11,368:INFO:lightgbm==3.2.1
2022-02-25 14:47:11,778:INFO:catboost==1.0.4
2022-02-25 14:47:12,004:INFO:mlflow==1.20.2
2022-02-25 14:47:12,004:INFO:Checking Exceptions
2022-02-25 14:47:12,004:INFO:Declaring global variables
2022-02-25 14:47:12,004:INFO:USI: ed55
2022-02-25 14:47:12,004:INFO:pycaret_globals: {'target_param', 'imputation_regressor', 'pycaret_globals', 'stratify_param', 'data_before_preprocess', 'html_param', 'experiment__', 'fix_imbalance_method_param', 'gpu_param', '_available_plots', '_ml_usecase', '_internal_pipeline', 'fold_shuffle_param', 'X', 'prep_pipe', 'n_jobs_param', 'fix_imbalance_param', 'USI', 'transform_target_method_param', 'fold_param', 'exp_name_log', '_all_models_internal', 'seed', 'fold_generator', 'log_plots_param', 'create_model_container', 'logging_param', 'display_container', 'master_model_container', 'iterative_imputation_iters_param', '_gpu_n_jobs_param', 'transform_target_param', '_all_models', 'fold_groups_param', '_all_metrics', 'imputation_classifier'}
2022-02-25 14:47:12,006:INFO:Preparing display monitor
2022-02-25 14:47:12,007:INFO:Preparing display monitor

Code Snippet:

**Setting up and Assigning :**

set_data = setup(data = df, silent = True, preprocess=False,session_id=123)
model = create_model(model='lof',contamination=0.20,n_neighbors=4, n_jobs=32)
print(model)
 o_df = assign_model(lof_model)
 
 **Predicting:**
 pred = predict_model(model, data = df_test)

**Comment by moezali1** [comment]
@praveenmec67 `assign_model` function uses decision score to label the anomalies whereas `predict_model` function uses trained model to score new data points. 

https://pyod.readthedocs.io/en/latest/api_cc.html
